src/nunit.analyzers/SimplifyValues/SimplifyValuesAnalyzer.cs

```diff
@@ -0,0 +1,192 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.Extensions;
+
+namespace NUnit.Analyzers.SimplifyValues;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class SimplifyValuesAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly DiagnosticDescriptor simplifyValues = DiagnosticDescriptorCreator.Create(
+        id: AnalyzerIdentifiers.SimplifyValues,
+        title: SimplifyValuesAnalyzerConstants.SimplifyValuesTitle,
+        messageFormat: SimplifyValuesAnalyzerConstants.SimplifyValuesMessage,
+        category: Categories.Style,
+        defaultSeverity: DiagnosticSeverity.Info,
+        description: SimplifyValuesAnalyzerConstants.SimplifyValuesDescription);
+
+    private static readonly ImmutableHashSet<string> nonCombinatorialAttributes = ImmutableHashSet.Create("SequentialAttribute", "PairwiseAttribute");
+
+    /// <summary>
+    /// Types for which NUnit's ValuesAttribute can deduce all values.
+    /// </summary>
+    private enum HandledType
+    {
+        Boolean,
+        NullableBoolean,
+        Enum,
+        NullableEnum,
+        NotHandled,
+    }
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(simplifyValues);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.EnableConcurrentExecution();
+        context.RegisterCompilationStartAction(AnalyzeCompilationStart);
+    }
+
+    private static void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+    {
+        var valuesType = context.Compilation.GetTypeByMetadataName(NUnitFrameworkConstants.FullNameOfTypeValuesAttribute);
+        if (valuesType is null)
+            return;
+
+        context.RegisterSymbolAction(symbolContext => AnalyzeParameter(symbolContext, valuesType), SymbolKind.Parameter);
+    }
+
+    private static void AnalyzeParameter(SymbolAnalysisContext symbolContext, INamedTypeSymbol valuesType)
+    {
+        var parameterSymbol = symbolContext.Symbol as IParameterSymbol;
+        if (parameterSymbol is null)
+            return;
+
+        var methodSymbol = parameterSymbol.ContainingSymbol as IMethodSymbol;
+        if (methodSymbol is null)
+            return;
+
+        var methodAttributes = methodSymbol.GetAttributes();
+        var methodHasNonCombinatorialAttribute = methodAttributes
+            .Any(attribute => attribute.AttributeClass is { } attributeClass
```

Comment:
I find `is { }` very hard to read. 
```suggestion
            .Any(attribute => attribute.AttributeClass is INamedTypeSymbol attributeClass
```